<?xml version="1.0"?>

<launch>


<param name="Rb" type="double" value="3.5"/>
<param name="d" type="double" value="10.0"/>
<param name="c" type="double" value="36.0"/>
<param name="ref_vel" type="double" value="1.0"/>


<!--Run the stage simulator-->
<node pkg = "stage_ros" name = "stageros" type = "stageros" output = "screen" args="-d $(find example)/worlds/map_4-2-3-1-1.world">
</node>

<!-- Communication manager args="N_robots"-->
<node pkg = "example" name = "communication" type = "communication.py" args="0 0 0 0 1 1 2 2 2 3 4" output="screen">
</node>



<!--Run the cntroller node args="robot_number robot_group" -->
<group ns = "robot_0">
    <node pkg = "example" name = "controller" type = "dubins-seg.py" args="0 0 20.000 0.000 0.000 90.000" output="screen">
    </node>
</group>
<group ns = "robot_1">
    <node pkg = "example" name = "controller_1" type = "dubins-seg.py" args="1 0 28.284 28.284 0.000 135.000" output="screen">
    </node>
</group>
<group ns = "robot_2">
    <node pkg = "example" name = "controller_2" type = "dubins-seg.py" args="2 0 -30.000 0.000 0.000 90.000" output="screen">
    </node>
</group>
<group ns = "robot_3">
    <node pkg = "example" name = "controller_3" type = "dubins-seg.py" args="3 0 1.000 10.000 0.000 0.000" output="screen">
    </node>
</group>
<group ns = "robot_4">
    <node pkg = "example" name = "controller_4" type = "dubins-seg.py" args="4 1 10.000 0.071 0.000 -90.000" output="screen">
    </node>
</group>
<group ns = "robot_5">
    <node pkg = "example" name = "controller_5" type = "dubins-seg.py" args="5 1 0.000 -30.000 0.000 180.000" output="screen">
    </node>
</group>
<group ns = "robot_6">
    <node pkg = "example" name = "controller_6" type = "dubins-seg.py" args="6 2 -3.473 19.696 0.000 190.000" output="screen">
    </node>
</group>
<group ns = "robot_7">
    <node pkg = "example" name = "controller_7" type = "dubins-seg.py" args="7 2 -10.000 0.001 0.000 90.000" output="screen">
    </node>
</group>
<group ns = "robot_8">
    <node pkg = "example" name = "controller_8" type = "dubins-seg.py" args="8 2 0.000 -20.000 0.000 0.000" output="screen">
    </node>
</group>
<group ns = "robot_9">
    <node pkg = "example" name = "controller_9" type = "dubins-seg.py" args="9 3 -40 0.000 0.000 -90.000" output="screen">
    </node>
</group>
<group ns = "robot_10">
    <node pkg = "example" name = "controller_9" type = "dubins-seg.py" args="10 4 0.000 30.000 0.000 0.000" output="screen">
    </node>
</group>


<!--Run the rviz for better visualization-->
<!-- <node pkg = "rviz" name = "rviz" type = "rviz" args="-d $(find example)/rviz/rviz_config.rviz" output="screen">
</node> -->
<!-- <node pkg = "map_server" name = "map_server" type = "map_server" args="$(find example)/maps/map_1.yaml" output="screen">
</node> -->

</launch>
